src/ui/app/App.js

```diff
@@ -205,17 +206,12 @@ const RouteComponent = () => {
   const { COMPONENT_REPOSITORY_UI_ENABLED } = getRuntimeEnv();
   const contentMfe = useSelector(getMfeTargetContent);
 
-  const mfeRoutes = React.useMemo(() => contentMfe.reduce((acc, curr) => {
-    if (curr.paths) {
-      const routes = [];
-      curr.paths.forEach((route) => {
-        routes.push({ route, id: curr.id });
-      });
-      return [...acc, ...routes];
-    }
-    return acc;
+  const mfeRoutes = React.useMemo(
+    () => generateMfeRoutes(contentMfe),
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []), [JSON.stringify(contentMfe)]);
+    [],
```

Comment:
should it be removed? looks like you are passing 3 parameters to `useMemo` instead of 2.